geode-core/src/main/java/org/apache/geode/distributed/internal/InternalClusterConfigurationService.java

```diff
@@ -834,7 +834,11 @@ public CacheConfig getCacheConfig(String group) {
     if (group == null) {
       group = CLUSTER_CONFIG;
     }
-    String xmlContent = getConfiguration(group).getCacheXmlContent();
+    Configuration groupConfig = getConfiguration(group);
```

Comment:
this is probably not related to your PR, but if you make this change here, the updateCacheConfig in this class needs to be changed as well to handle the null value returned by this method.